### PR TITLE
#5509: Add ternary ops to ttnn

### PR DESCRIPTION
**Pull request by Aswinmcw**
Add support for ternary ops to ttnn
Sweep Test Results:
[addcdiv.csv](https://github.com/tenstorrent-metal/tt-metal/files/14356394/addcdiv.csv)
[addcmul.csv](https://github.com/tenstorrent-metal/tt-metal/files/14356395/addcmul.csv)
[arange.csv](https://github.com/tenstorrent-metal/tt-metal/files/14356396/arange.csv)
[mac.csv](https://github.com/tenstorrent-metal/tt-metal/files/14356399/mac.csv)
[where.csv](https://github.com/tenstorrent-metal/tt-metal/files/14356400/where.csv)

### CODE REVIEW

**Comment by Aswinmcw** [comment]
Slow Dispatch Test: https://github.com/tenstorrent-metal/tt-metal/actions/runs/7985990013
Fast Dispatch Test: https://github.com/tenstorrent-metal/tt-metal/actions/runs/7985988863

**Review by arakhmati — CHANGES_REQUESTED** [review]
`arange` needs to be moved to `creation.py`. And it must have a mandatory `device` argument and only run on device

File: ttnn/ttnn/operations/ternary.py [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+
+        name_to_torch_function = {
+            "mac": torch_mac,
+        }
+        torch_function = name_to_torch_function[name]
+        input_tensor = ttnn.to_torch(input_tensor)
+        return torch_function(input_tensor)
+
+    def _ternary_validate_input_tensors(operation_name, input_tensor, *args, **kwargs):
+        ttnn.validate_input_tensor(
```
Comment (arakhmati) [inline ttnn/ttnn/operations/ternary.py:35]: There must be ttnn.validate_input_tensor for all 3 tensors

**Review by arakhmati — CHANGES_REQUESTED** [review]

**Comment by Aswinmcw** [comment]
Modified the following:
Moved arange from ternary to creation
Added device argument for arange op
Added tensor validation condition for all inputs

**Review by arakhmati — APPROVED** [review]

**Pull request merged**